src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java

```diff
@@ -139,6 +144,34 @@ public void writeStaticContent() throws Exception {
     assertThat(content, is(HELLO_WORLD));
   }
 
+  @Test
+  public void writeWithLock() throws Exception {
+    testHarness.makeDir(TEMP_DIRECTORY);
+    String path = Paths.get(testHarness.getWorkingDirectory(), TEMP_DIRECTORY, "test.txt").toString();
+    doWrite("writeWithLock", path, HELLO_WORLD, CREATE_NEW, false);
+
+    String content = toString(readPath(path).getPayload().getValue());
+    assertThat(content, is(HELLO_WORLD));
+  }
+
+  //TODO: MULE-16515 ignore this test until issue is fixed.
+  @Test
+  @Ignore
+  public void writeOnLockedFile() throws Exception {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Fix or remove this skipped unit test ([squid:S1607](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1607))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdg&open=AWkMI-sJotIXnmsUtDdg)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdg